addons/mail/static/src/core/web/recipients_input.js

```diff
@@ -0,0 +1,304 @@
+            onRecordSaved: (partner) => {
+                const recipients = this.getAllMailThreadRecipients().filter(
+                    recipient => recipient.partner_id === partner.resId
+                );
+                for (const recipient of recipients) {
+                    Object.assign(recipient, {
+                        email: partner.data.email,
+                        name: partner.data.name,
+                    });
+                }
+            }
+        });
+
+        this.openListViewToSelectResPartner = useSelectCreate({
+            resModel: "res.partner",
+            activeActions: {
+                create: false,
+                link: true, // Unable multi-select
+            },
+            /** @param {Object} resIds */
+            onSelected: async (resIds) => {
+                const partners = await this.orm.searchRead(
+                    "res.partner",
+                    [["id", "in", Array.from(resIds)]],
+                    ["email", "id", "lang", "name"]
+                );
+                for (const partner of partners) {
+                    this.insertAdditionalRecipient({
+                        email: partner.email,
+                        lang: partner.lang,
+                        name: partner.name,
+                        partner_id: partner.id,
+                        persona: { type: "partner", id: partner.id },
+                    });
+                }
+            },
+        });
+    }
+
+    getAutoCompleteSources() {
+        return [
+            {
+                placeholder: _t("Loading..."),
+                /** @param {string} term */
+                options: async (term) => {
+                    const partnerIds = new Set();
+                    const recipients = this.getAllMailThreadRecipients();
+
+                    for (const recipient of recipients) {
+                        if (recipient.partner_id) {
+                            partnerIds.add(recipient.partner_id);
+                        }
+                    }
+
+                    const options = [];
+                    const [name, email] = term ? parseEmail(term) : ["", ""];
+
+                    const limit = 8;
+                    const matches = await this.orm.searchRead(
+                        "res.partner",
+                        [
+                            ["id", "not in", Array.from(partnerIds)],
+                            "|",
+                            ["name", "ilike", term],
+                            ["email", "ilike", term],
+                        ],
+                        ["email", "id", "lang", "name"],
+                        { limit }
+                    );
+
+                    options.push(
+                        ...matches.map((match) => ({
+                            id: match.id,
+                            label: match.email
+                                ? _t("%(partner_name)s <%(partner_email)s>", {
```

Comment:
Meh. If you want to introduce your own ``formataddr``, you should probably also quote content of partner_name (name may contain quotes, ...).